include/ibamr/vc_ins_utilities.h

```diff
@@ -0,0 +1,398 @@
+     * Set the viscosity based on the current level set information for two-phase flows.
+     */
+    void setViscosityPatchData2PhaseFlows(int mu_idx,
+                                          SAMRAI::tbox::Pointer<SAMRAI::hier::Variable<NDIM> > mu_var,
+                                          SAMRAI::tbox::Pointer<IBTK::HierarchyMathOps> hier_math_ops,
+                                          const int cycle_num,
+                                          const double time,
+                                          const double current_time,
+                                          const double new_time);
+
+    /*!
+     * Set the viscosity based on the current level set information for three-phase flows.
+     */
+    void setViscosityPatchData3PhaseFlows(int mu_idx,
+                                          SAMRAI::tbox::Pointer<SAMRAI::hier::Variable<NDIM> > mu_var,
+                                          SAMRAI::tbox::Pointer<IBTK::HierarchyMathOps> hier_math_ops,
+                                          const int cycle_num,
+                                          const double time,
+                                          const double current_time,
+                                          const double new_time);
+
+    /*!
+     * Name of this object.
+     */
+    std::string d_object_name;
+
+    /*!
+     * Pointer to advection-diffusion solver.
+     */
+    SAMRAI::tbox::Pointer<IBAMR::AdvDiffHierarchyIntegrator> d_adv_diff_solver;
+
+    /*!
+     * Level set variable.
+     */
+    SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > d_ls_gas_var, d_ls_solid_var;
+
+    /*!
+     * Density.
+     */
+    double d_rho_liquid, d_rho_gas, d_rho_solid;
+
+    /*!
+     * Viscosity.
+     */
+    double d_mu_liquid, d_mu_gas, d_mu_solid;
+
+    /*!
+     * Level set reinitialization interval.
+     */
+    int d_ls_reinit_interval;
+
+    /*!
+     * Number of interface cells over which to smooth the material properties.
+     */
+    double d_num_gas_interface_cells, d_num_solid_interface_cells;
+
+    /*!
+     * Whether or not to set viscosity in the solid region.
+     */
+    bool d_set_mu_solid;
+
+    /*!
+     * Number of phases. Valid options are 2 and 3.
+     */
+    int d_num_phases = 2;
+
+}; // SetFluidProperties
+
+class GravityForcing : public IBTK::CartGridFunction
+{
+public:
+    /*!
+     * \brief Class constructor.
+     */
+    GravityForcing(const std::string& object_name,
+                   SAMRAI::tbox::Pointer<INSVCStaggeredHierarchyIntegrator> ins_hierarchy_integrator,
+                   std::vector<double> grav_const,
+                   std::string grav_type = "FULL",
+                   SAMRAI::tbox::Pointer<AdvDiffHierarchyIntegrator> adv_diff_hierarchy_integrator = nullptr,
+                   SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > ls_gas_var = nullptr,
+                   SAMRAI::tbox::Pointer<SAMRAI::tbox::Database> input_db = nullptr);
```

Comment:
I have two different constructors now. Since the ```grav_type``` option is unique for each constructor, I removed ```enum``` for ```grav_type``` and set it in the constructor directly.